python/setup.py

```diff
@@ -30,11 +26,18 @@ def run(self):
             "core/src/global_scheduler/global_scheduler"]
 }
 
+
+class BinaryDistribution(Distribution):
+  def has_ext_modules(foo):
+    return True
+
+
 setup(name="ray",
       version="0.1.0",
       packages=find_packages(),
       package_data=package_data,
-      cmdclass={"install": install},
+      cmdclass={"build_ext": build_ext},
```

Comment:
Looks like we also need to include a line like

```
# Dummy extension to trigger build_ext
ext_modules=[Extension("__dummy__", sources=[])],
```

as an argument to `setup` in order to trigger the `build_ext` to run. Note I found this in https://github.com/apache/arrow/blob/master/python/setup.py#L357.

It still doesn't solve the problem in which `build_ext` seems to run after the package data gets included (so you still need to run things twice).